index.js

```diff
@@ -1,31 +1,47 @@
-/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
-   Complete the below for code reviewers' convenience:
-
-   Code repository: _put repo URL here_
-   Web app: _put project's github pages URL here_
-   */
-
-// ======== OBJECTS DEFINITIONS ========
-// Define your objects here
-
-
-// ======== OUTPUT ========
-/* Use print(message) for output.
-   Default tag for message is <pre>. Use print(message,'div') to change containing element tag.
-
-   Message can contain HTML markup. You may also tweak index.html and/or styles.css.
-   However, please, REFRAIN from improving visuals at least until your code is reviewed
-   so code reviewers might focus on a single file that is index.js.
-   */
-
-/* Print examples:
-   print('ABC');
-   print('<strong>ABC</strong>');
-   print('<strong>ABC</strong>', 'div');
-
-   print('human; John; male; 2; 2; Hello world!; Rex, Tom, Jenny');
-   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny');
-   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny', 'div');
-   */
-
-
+class Parent {
+  constructor(name, gender, legs, say) {
+    this.name = name
+    this.gender = gender
+    this.kind = this.constructor.name
+    this.legs = legs
+    this.say = say
+  }
+  get toString() {
+    return Object.keys(this)
+      .map(key => {
+        return this[key]
+      })
+      .join('; ')
+  }
+}
+
+class Dog extends Parent {
+  constructor(name, gender) {
+    super(name, gender, 4, `woof - woof`)
+  }
+}
+
+class Cat extends Parent {
+  constructor(name, gender) {
+    super(name, gender, 4, `meow - meow`)
+  }
+}
+class Human extends Parent {
+  constructor(name, gender) {
+    super(name, gender, 2, `Hello, I'm ${name}`)
+  }
+  get toString() {
+    return `${super.toString}; 2 hands`
```

Comment:
With explicit list of properties in `Parent.toString` you can safely add extra properties in child classes and use those.

`[super.toString(), this.hands + ' hands'].join('; ');` will do.